wordcount/count/FileOpenHandleRuneBenchmark.cs

```diff
@@ -0,0 +1,62 @@
+using System.Buffers;
+using System.Linq.Expressions;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text;
+using BenchmarkData;
+
+namespace FileOpenHandleRuneBenchmark;
+
+public static class FileOpenHandleRuneBenchmark
+{
+    public static Count Count(string path)
+    {
+        int wordCount = 0, lineCount = 0, byteCount = 0;
+        bool wasSpace = true;
+
+        byte[] buffer = ArrayPool<byte>.Shared.Rent(BenchmarkValues.Size);
+        using var handle = File.OpenHandle(path, FileMode.Open, FileAccess.Read, FileShare.Read, FileOptions.SequentialScan);
+        int index = 0;
+
+        // Read content in chunks, in buffer, at count lenght, starting at byteCount
+        int count = 0;
+        while ((count = RandomAccess.Read(handle, buffer.AsSpan(index), byteCount)) > 0 || index > 0)
+        {
+            byteCount += count;
+            Span<byte> bytes = buffer.AsSpan(0, count + index);
+
+            while (bytes.Length > 0)
+            {
+                var status = Rune.DecodeFromUtf8(bytes, out Rune rune, out int bytesConsumed);
+
+                // bad read due to low buffer length
+                if (status is not OperationStatus.Done && bytes.Length < 4)
```

Comment:
Bug: This will cause data loss on invalid UTF-8 and will tear reads for valid UTF-8. Consider instead:

```cs
if (status == OperationStatus.NeedMoreData && count > 0)
{
    bytes[..bytesConsumed].CopyTo(buffer);
    index = bytesConsumed;
    break;
}
```

(Note: The `&& count > 0` condition is necessary to avoid this method from entering an infinite loop!)

No need to special-case `InvalidData`. It can be treated equivalently to `Done`.